contracts/src/token/erc20/extensions/flashmint.rs

```diff
@@ -0,0 +1,164 @@
+
+use alloy_primitives::{Address, U256, Bytes, B256, b256};
+use stylus_sdk::msg;
+use stylus_sdk::{
+    contract, prelude::*,
+     call::{ Call, MethodError}
+};
+use crate::token::erc20::{self, Erc20, IErc20};
+use crate::flashloan::borrower::IERC3156FlashBorrower;
+use alloy_sol_types::sol;
+
+sol! {
+
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC3156UnsupportedToken(address token);
+
+
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC3156ExceededMaxLoan(uint256 maxLoan);
```

Comment:
```suggestion
    error ERC3156ExceededMaxLoan(uint256 max_loan);
```
Using snake_case is the convention for both events and errors in Stylus